src/error.rs

```diff
@@ -23,12 +23,12 @@ pub type Result<T> = std::result::Result<T, Error>;
 /// An error that can occur in the `mongodb` crate. The inner
 /// [`ErrorKind`](enum.ErrorKind.html) is wrapped in an `Arc` to allow the errors to be
 /// cloned.
-#[derive(Clone, Debug, Error)]
+#[derive(Debug, Error)]
 #[error(display = "{}", kind)]
 #[non_exhaustive]
 pub struct Error {
     /// The type of error that occurred.
-    pub kind: Arc<ErrorKind>,
+    pub kind: ErrorKind,
```

Comment:
This is the only substantive change in this commit; the rest are just fixes for the Arc being removed